components/button/Button.spec.tsx

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { render, fireEvent, screen } from '@testing-library/react'
+
+// Components
+import { Button } from './Button'
+
+describe('Button', () => {
+  it('executes the action when button is click', () => {
+    const onClick = jest.fn()
+    render(<Button label="save" onClick={onClick} />)
+
+    fireEvent.click(screen.getByText('save'))
+
+    expect(onClick).toHaveBeenCalled()
+  })
+})
```

Comment:
Have you considered this? https://storybook.js.org/docs/react/workflows/unit-testing

As it turns out, you can easily do this to reuse data in stories:

```javascript
//👇 Imports a specific story for the test
import { Primary } from './Button.stories';

it('renders the button in the primary state', () => {
  render(<Primary {...Primary.args} />);
  expect(screen.getByRole('button')).toHaveTextContent('Primary');
});
```

Reply:
Interesting, I didn't know you could that.

But I think we will mainly cover testing with e2e tests, and some specific parts will be covered by unit tests, and unit tests tend to test logic more than the UI. Also, for tests, you want different data for different scenarios. So, not sure what's the benefit here. 🤔